### PR TITLE
Add pull request check for linked issues to CI workflow

### LINKED ISSUE #3971: Add GitHub actions to check if pull request has linked GitHub issue

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #3971

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: .github/workflows/pull_request_check.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  pull_request_check:
+    name: pull request check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: nearform/github-action-check-linked-issues@v1
+        id: check-linked-issues
+        with:
+          exclude-branches: "release_v/**, backport_v/**, main, latest-dep-update-/**, min-dep-update-/**, dependabot/**"
+          comment: false
+          github-token: ${{ secrets.REPO_SCOPED_TOKEN }}
```
Comment (gsheni) [inline .github/workflows/pull_request_check.yaml:15]:
```suggestion
          github-token: ${{ secrets.MACHINEFL_DEPENDENCY_CHECKER_TOKEN }}
```

**Review by thehomebrewnerd — APPROVED** [review]

**Review by jeremyliweishih — APPROVED** [review]
LGTM thanks!

**Comment by gsheni** [comment]
The fix for `Build conda package` can be found here: https://github.com/alteryx/evalml/pull/3978

**Pull request merged**